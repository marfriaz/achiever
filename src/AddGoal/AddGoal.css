.AddGoal {
  color: #dedede;
}

.AddGoal h2 {
  margin-left: 16px;
}
