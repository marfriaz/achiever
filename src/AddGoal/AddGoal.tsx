import React, { useContext, useState } from "react";
import { useHistory } from "react-router-dom";
import AchieverForm from "../AchieverForm/AchieverForm";
import ApiContext from "../ApiContext";
import config from "../config";
import "./AddGoal.css";

export default function AddGoal() {
  interface stateInterface {
    inputValue: string;
  }

  const [state, setState] = useState<stateInterface>();

  const history = useHistory();

  const { addGoal } = useContext(ApiContext)!;

  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
    event.preventDefault();
    const goal = {
      name: state,
    };
    fetch(`${config.API_ENDPOINT}/api/goals`, {
      method: "POST",
      headers: {
        "content-type": "application/json",
      },
      body: JSON.stringify(goal),
    })
      .then((res) => {
        if (!res.ok) {
          return res.json().then((e) => Promise.reject(e));
        }
        return res.json();
      })
      .then((goal) => {
        addGoal(goal);
        history.push(`/goal/${goal.id}`);
      })
      .catch((error) => {
        console.error({ error });
      });
  };

  const handleChange = (event: React.FormEvent) => {
    var inputValue = (event.target as any).value;
    setState(inputValue);
  };

  return (
    <section className="AddGoal">
      <h2>Create a goal</h2>
      <AchieverForm onSubmit={handleSubmit}>
        <div className="field">
          <label htmlFor="goal-name-input">Name</label>
          <input
            type="text"
            id="goal-name-input"
            name="goal-name"
            onChange={(ev) => handleChange(ev)}
          />
        </div>
        <div className="buttons">
          <button type="submit">Add goal</button>
        </div>
      </AchieverForm>
    </section>
  );
}
