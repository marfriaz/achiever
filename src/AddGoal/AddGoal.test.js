import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AddGoal from "./AddGoal";

describe(`AddItemForm component`, () => {
  it("renders the complete form", () => {
    const wrapper = shallow(<AddGoal />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
