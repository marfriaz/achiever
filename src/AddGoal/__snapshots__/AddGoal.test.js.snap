// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemForm component renders the complete form 1`] = `
<section
  className="AddGoal"
>
  <h2>
    Create a goal
  </h2>
  <AchieverForm>
    <div
      className="field"
    >
      <label
        htmlFor="goal-name-input"
      >
        Name
      </label>
      <input
        id="goal-name-input"
        type="text"
      />
    </div>
    <div
      className="buttons"
    >
      <button
        type="submit"
      >
        Add goal
      </button>
    </div>
  </AchieverForm>
</section>
`;
